### PR TITLE
Ramda refact

### PR DESCRIPTION
**Pull request by admin12345654321**
#### What's this PR do?

refactors app

#### Sanity Check:
- [x] Tested on IE 11.
- [x] Tested on Chrome.
- [x] Tested on Firefox.
- [ ] Unit tests

### CODE REVIEW

File: src/common/hooks/index.js [hunk 1]
```diff
@@ -1,16 +1,61 @@
...
+    const { id, body: content, type, originalIndex } = newPanel
+    const panelId = randomKey(10)
+
+    setPanels([...panels, { id, content, type, originalIndex, key: panelId }])
+    onAddPanel({ key: panelId, id, content, originalIndex })
+  }
+
+  const destroyPanel = meta => {
+    const destroy = item => item.key !== meta.panelId
+    const newState = filter(destroy, panels)
```
Comment (cristian-sandu) [inline src/common/hooks/index.js]: // have a look at the **whereEq** function
```
const newState = reject(whereEq({
  key: meta.panelId
}), panels)
```
OR:
```
const whereNotEq = complement(whereEq)
const newState = filter(whereNotEq({
  key: meta.panelId
}), panels)
```

File: src/common/components/Grid/index.js [hunk 2]
```diff
@@ -1,11 +1,10 @@
 import React from 'react'
 import { AgGridReact } from 'ag-grid-react'
+import { calcCellSize } from '../../utils'
 import './Grid.scss'
 
 const Grid = props => {
-  const cellSize = {
-    fontSize: props.rowHeight / 2,
-  }
+  const cellSize = calcCellSize(props.rowHeight)
```
Comment (newalex997) [inline src/common/components/Grid/index.js]: `const cellSize = getCellFontSize(props.rowHeight)`

File: src/App/utils.js [hunk 3]
```diff
@@ -12,7 +13,7 @@ export const onDropPanel = (panels, callback) => ({
   accept: 'panel',
   drop: ({ panelId }) => {
     const panel = panelTypes.find(panel => panel.id === panelId)
-    const panelsCount = panels.length
+    const panelsCount = length(panels)
```
Comment (newalex997) [inline src/App/utils.js]: ramda is not required here

File: src/App/utils.js [hunk 4]
```diff
@@ -49,26 +50,21 @@ export const onMovePanel = (panels, callback) => (id, overId) => {
...
-// move it in a separate file
-// const DEFAULT_PREFIX = 'CLIENT'
-
-// const reducer = (context, prefix) => (acc, value) => {
-//   const nextAcc = { ...acc }
-
-//   nextAcc[value] = `${prefix}/${context}.${value}`
+export const regeneratePanels = (panels, panelTypes) => {
+  return values(panels).map(panel => {
+    const panelDetails = panelTypes.find(item => item.id === panel.details.id)
```
Comment (newalex997) [inline src/App/utils.js]: `const panelDetails = find(propEq('id', panel.details.id), panelTypes)`

File: src/common/selectors/index.js [hunk 5]
```diff
@@ -1,34 +1,32 @@
...
+import { defaultTo, path } from 'ramda'
+import {
+  REDUCERS_FIELDS,
+  CRYPTOCURRENCIES,
+  CURRENCIES,
+  REDUCERS,
+} from '../../constants'
+
+const getPanelProp = (prop, type = REDUCERS.DATA) => (store, panel) => {
+  if (path([type, panel.meta.panelId], store))
```
Comment (newalex997) [inline src/common/selectors/index.js]: without if just return patch => `return store[type][panel.meta.panelId][prop]`

File: src/common/hooks/index.js [hunk 6]
```diff
@@ -1,16 +1,60 @@
...
+  panelItems,
+) => {
+  const [panels, setPanels] = useState([])
+
+  const regeneratedPanels = useCallback(() => {
+    setPanels(regeneratePanels(panelItems, panelTypes))
+  }, [panelItems, regeneratePanels])
+
+  useEffect(() => {
+    regeneratedPanels()
```
Comment (newalex997) [inline src/common/hooks/index.js]: move `setPanels` directly here

**Pull request merged**